### PR TITLE
refactor: Separating statement parse func

### PR DESCRIPTION
**Pull request by Gump9**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
split `show` ,`describe`,`explain`,`drop`statement parsing into separate *_parser.rs file and place it in the sql/parsers directory, just like other statements.
## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
close https://github.com/GreptimeTeam/greptimedb/issues/1914

### CODE REVIEW

**Comment by Gump9** [comment]
I only refactor the `show` statement parsing now,and i want to know if do the right thing?If i do,i will continue refactoring the left statement parsing,thx!

**Comment by MichaelScofield** [comment]
@Gump9 you're doing it right. Just delete codes instead of comment them out.

**Comment by Gump9** [comment]
@waynexia Hi,I have done this refactor😈

**Review by waynexia — APPROVED** [review]
LGTM

File: src/sql/src/parser.rs [hunk 1]
```diff
@@ -351,276 +177,19 @@ impl<'a> ParserContext<'a> {
...
-            },
-            _ => self.unsupported(self.peek_token_as_string()),
-        }
-    }
 }
 
 #[cfg(test)]
 mod tests {
-    use std::assert_matches::assert_matches;
+    // use std::assert_matches::assert_matches;
```
Comment (MichaelScofield) [inline src/sql/src/parser.rs:184]:
```suggestion
```

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**